attachments_component/site/src/Model/AttachmentsModel.php

```diff
@@ -387,15 +391,19 @@ public function &getAttachmentsList()
 		$query = $db->getQuery(true);
 		$query->select('a.*, u.name as creator_name')->from('#__attachments AS a');
 		$query->leftJoin('#__users AS u ON u.id = a.created_by');
-
+		if ($attachmentid != null) {
+			$query->where('a.id in (' . $attachmentid . ')' );
+			}
 		if ( $parent_id == 0 ) {
 			// If the parent ID is zero, the parent is being created so we have
 			// do the query differently
 			$user_id = $user->get('id');
 			$query->where('a.parent_id IS NULL AND u.id=' . (int)$user_id);
 			}
 		else {
-			$query->where('a.parent_id='.(int)$parent_id);
+			if ($parent_id != '%') {
```

Comment:
@JLTRY I cannot understand what you want to achieve here. I can see that you set parent_id to % earlier but since you check if parent isn't % why did you change the query from "a.parent_id=" to "a.parent_id like"? Did you want to add attachments from other articles maybe?

Reply:
yes sorry the query it should not be changed for a.parent_id. Yes if attachmentsid is not null it is of format xx,yy,zz. YEs I want to be able to add attachments from everywhere